zebra-state/src/service.rs

```diff
@@ -491,6 +485,10 @@ impl StateService {
             }
 
             self.drain_queue_and_commit_finalized();
+
+            if let Some(finalized_tip_height) = self.read_service.db.finalized_tip_height() {
+                self.sent_blocks.prune_by_height(finalized_tip_height);
```

Comment:
`prune_by_height` calls `finish_batch`. Doesn't that imply that batches are effectively single blocks?

Reply:
It doesn't. Sorry for the confusion.